client/packages/purchasing/src/purchase_order/api/hooks/usePurchaseOrderList.ts

```diff
@@ -55,9 +58,21 @@ export const usePurchaseOrderList = (queryParams: ListParams) => {
     nodes: PurchaseOrderRowFragment[];
     totalCount: number;
   }> => {
-    const filter = {
-      ...filterBy,
-    };
+    const filter = { ...filterBy } as PurchaseOrderFilterInput;
+
+    if (filter?.confirmedDatetime?.equalTo) {
+      const confirmedDate = filter.confirmedDatetime;
+      confirmedDate.equalTo = Formatter.toIsoString(
+        DateUtils.getDateOrNull(confirmedDate.equalTo)
+      );
+    }
+
+    if (filter?.sentDatetime?.equalTo) {
+      const sentDate = filter.sentDatetime;
+      sentDate.equalTo = Formatter.toIsoString(
+        DateUtils.getDateOrNull(sentDate.equalTo)
+      );
+    }
```

Comment:
Not needed the component should already be doing this

```suggestion
    const filter = { ...filterBy };
```